app/resources/file.py

```diff
@@ -173,14 +248,23 @@ def create_file(data: dict, user: str) -> dict:
 
     filename: str = data["filename"]
     content: str = data["content"]
+    is_directory: bool = data["is_directory"]
+    parent_dir_uuid: str = data["parent_dir_uuid"]
 
     file_count: int = wrapper.session.query(func.count(File.uuid)).filter_by(
-        device=device.uuid, filename=filename
+        device=device.uuid, filename=filename, parent_dir_uuid=parent_dir_uuid
     ).scalar()
 
     if file_count > 0:
         return file_already_exists
 
-    file: File = File.create(device.uuid, filename, content)
+    parent_dir: File = wrapper.session.query(File).filter_by(device=device.uuid, parent_dir_uuid=parent_dir_uuid)
+    if not parent_dir:
```

Comment:
You also need to check if the parent is actually a directory:
```suggestion
    if not (parent_dir and parent_dir.is_directory):
```